matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/pushrules/PushRuleService.kt

```diff
@@ -51,11 +51,7 @@ interface PushRuleService {
 //    fun fulfilledBingRule(event: Event, rules: List<PushRule>): PushRule?
 
     interface PushRuleListener {
-        fun onMatchRule(event: Event, actions: List<Action>)
-        fun onRoomJoined(roomId: String)
-        fun onRoomLeft(roomId: String)
-        fun onEventRedacted(redactedEventId: String)
-        fun batchFinish()
+        fun onEvents(pushEvents: PushEvents)
```

Comment:
using a single function here also allows up to simplfy the notification updates as we can batch all the events into one update pass, reducing the amount of synchronisation locks and redundant render passes

we might not need the `private var firstThrottler = FirstThrottler(200)` anymore because of how few individual updates we make!

Reply:
OK, You should add a file for the changelog with `.removal` extension to notify SDK users about this API break.

Reply:
added 88fbb8f8fdda542b93d4d052b921ab4ab5cc668c